File: BE/src/main/java/Project/AirFlowMonitor/model/Office.java
package Project.AirFlowMonitor.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({"employees", "sensorOffices"})
public class Office {
    @EmbeddedId
    private OfficeId id;
    private double surface;
    private int capacity;
    @OneToMany(mappedBy = "office")
    private List<Employee> employees;

}

File: FE/my-app/src/model/office.js
class Office {
    constructor(id, surface, capacity, employees, sensorOffices) {
      this.id = id || new Id();
      this.surface = surface;
      this.capacity = capacity;
      this.employees = employees || []; 
      this.sensorOffices = sensorOffices || [];
    }
}

class Id {
  constructor(buildingId, officeId) {
    this.buildingId = buildingId;
    this.officeId = officeId;
  }
}